lib/jekyll/layout.rb

```diff
@@ -43,7 +45,11 @@ def initialize(site, base, name)
     #
     # Returns nothing.
     def process(name)
-      self.ext = File.extname(name)
+      self.mime_type = MIME::Types.find{|t| ".#{t.preferred_extension}" == File.extname(name)}
```

Comment:
Future proofing, really, and avoiding a partial solution when a better one exists. If `mime_types` was a new dependency I'd think twice, but as it's already in the bundle, no harm in using it.